nf_core/lint/nfcore_yml.py

```diff
@@ -0,0 +1,74 @@
+import os
+import re
+
+from nf_core import __version__
+
+REPOSITORY_TYPES = ["pipeline", "modules"]
+
+
+def nfcore_yml(self):
+    """Repository ``.nf-core.yml`` tests
+
+    The ``.nf-core.yml`` contains metadata for nf-core tools to correctly apply its features.
+
+    * repository type:
+
+        * Check that the repository type is set.
+
+    * nf core version:
+
+         * Check if the nf-core version is set to the latest version.
+
+    """
+    passed = []
+    warned = []
+    failed = []
+    ignored = []
+
+    # Remove field that should be ignored according to the linting config
+    ignore_configs = self.lint_config.get(".nf-core", [])
+
+    try:
+        with open(os.path.join(self.wf_path, ".nf-core.yml")) as fh:
+            content = fh.read()
+    except FileNotFoundError:
+        with open(os.path.join(self.wf_path, ".nf-core.yaml")) as fh:
+            content = fh.read()
```

Comment:
Not sure if this file name is ever written like that, it should be created by the tools, so no need for this extra check 🙂

Reply:
Was because I read this somewhere. https://github.com/nf-core/tools/blob/1148c4e3651845cfa9dcf2f5c67e20c3709e661c/nf_core/utils.py#L1011
https://github.com/nf-core/tools/blob/1148c4e3651845cfa9dcf2f5c67e20c3709e661c/CHANGELOG.md#L685
So i thought to keep it in there as some pipeline still might have the deprecated yaml.

Reply:
ah, good point. we leave it in then. 🙂 it seems to be just cutandrun for now: https://github.com/search?q=path%3A.nf-core.yaml&type=code